### ISSUE TITLE
How do you get the maximum electric field in a particular brain region

### ISSUE THREAD

**Issue by xiberJ** [simnibs/simnibs#126]
Dear developers, hello

First of all, thank you very much for your software. For the simulation results, how to obtain the electric field data of a specific area, such as the maximum electric field or minimum electric field data of a small area on the surface of the brain, and the electric field data of a region inside the brain (the maximum electric field and so on)?

**Comment by oulap** [comment]
You can do a region-of-interest-based analysis using either different atlases or just using the local region around a point. Here's an example of how to span a spherical ROI around a point and calculating the average electric file magnitude within that region:

https://simnibs.github.io/simnibs/build/html/tutorial/analysis.html

Similarly you could calculate the max on any percentile you want.

**Comment by xiberJ** [comment]
Thank you very much for your answer


If I need to take the maximum in that range,
Is ”mean_magnE = np.average(field[roi], weights=elm_vols[roi])
print('mean ', field_name, ' in M1 ROI: ', mean_magnE)“ in the code
Change to ”max_magnE = np.max(gm_surf.field[field_name][roi])
print('maximum ', field_name, ' in ', region_name, ': ', max_magnE)
“

**Comment by xiberJ** [comment]
Thank you very much for your answer

Can visualize and obtaining the inner regions of the brain of the electric field. In addition, I have seen some articles using this AAL-atlas to characterize the distribution and intensity of electric fields. How does this work?Through this study let me know a lot of interesting things, haha

**Comment by oulap** [comment]
You can extract the e-field from the mesh pretty much anywhere, you just need to define the region-of-interest. For deeper brain regions you could either look for a coordinate in MNI-space, map that to subject space, and then define a ROI around that coordinate, or you could use the structural segmentation inside the m2m_<subid>/segmentation-folder to define the ROI in subject space. I suggest you read through the analysis tutorial here:

https://simnibs.github.io/simnibs/build/html/tutorial/analysis.html

It will describe both coordinate based ROI creation and the surface-based atlas analysis.

**Comment by xiberJ** [comment]
Thank you for your generous advice, I have a basic understanding of this code, and also obtained the size of the field value of the specified area is greater than 99 percent; I would also like to ask if the simulated electric field results can be visualized by FreeSurfer

**Comment by oulap** [comment]
Yep, when you do the simulations you can choose to map the electric field results to different spaces. In a simulation script, if you set these fields:
```
s.map_to_surf = False
s.map_to_fsavg = False
s.map_to_vol = False
s.map_to_MNI = False
```
all to True instead of False, where s is the simulation structure, the results will get mapped to the central gray matter surface of the subject (map_to_surf), to the fsaverage surface (map_to_fsavg), to the input nifti volume space (map_to_vol), and to the MNI template space (map_to_mni). You can then visualize these fields in freeview either in volume or surface space, whichever you prefer. For example if you would want to visualize the electric field on the pial surface of a subject,  you would first open the surface file inside m2m_<subid>/surfaces/<hemi>.pial.gii as a surface in freeview, then in freeview select overlay -> load generic, and then choose a file inside simnibs_simulation/subject_overlays and load that.

**Comment by xiberJ** [comment]
![1681543728697](https://user-images.githubusercontent.com/122353886/232196370-f3a8e92e-dca6-4165-b4de-b90356b756ec.png)


thanks，
I also asked how do you look at the distribution of the electric field in a section like this

**Comment by oulap** [comment]
Not sure where that visualization is from, but please check this tutorial:

https://simnibs.github.io/simnibs/build/html/tutorial/visualization.html

We typically use gmsh to visualize the results, if you want to look at the distribution of the e-field over a section you should look into how to do clipping in gmsh (shown in the tutorial in the link above).
